### PR TITLE
Allow multiple (different) editors in one page

### PR DESCRIPTION
**Pull request by dgrammatiko**
Pull Request for Issue https://github.com/joomla/joomla-cms/issues/12555
### Steps to reproduce

Make sure that the default editor is tinyMCE

Edit `administrator/components/com_content/models/forms/article.xml` and add after 

``` xml
    <fieldset addfieldpath="/administrator/components/com_categories/models/fields" >
```

this:

``` xml
        <field
                name="test2"
                label ="Test Field"
                type="editor"
                width="300"
                filter="safehtml"
                editor="none"
                buttons="true"/>
        <field
                name="test3"
                label ="Test Field"
                type="editor"
                width="300"
                filter="safehtml"
                editor="codemirror"
                buttons="true"/>
```

Edit `administrator/components/com_content/views/article/tmpl/edit.php` and add after

``` html
                <fieldset class="adminform">
                    <?php echo $this->form->getInput('articletext'); ?>
                </fieldset>
```

this:

``` html
                <fieldset class="adminform">
                    <?php echo $this->form->getInput('test2'); ?>
                </fieldset>
                <fieldset class="adminform">
                    <?php echo $this->form->getInput('test3'); ?>
                </fieldset>
```

Then try to use the buttons to insert some images, contacts, etc. Oops it's broken for all except the last one.
### Summary of Changes

Introducing a simple API which is just some setters and some getters.
This is more informative:

``` js
// An object to hold each editor instance on page, only define if not defined.
Joomla.editors.instances = Joomla.editors.instances || {
        /**
         * Editors MUST register, per instance, the following:
         *
         * getValue         Type  Function
         *          Should return the complete data from the editor
         *          Example: function () { return this.element.value; }
         * setValue         Type  Function
         *          Should replace the complete data of the editor
         *          Example: function (text) { return this.element.value = text; }
         * replaceSelection  Type  Function
         *          Should replace the selected text of the editor
         *          If nothing selected, will insert the data at the cursor
         *          Example: function (text) { return insertAtCursor(this.element, text); }
         *
         * USAGE (jform_articletext is the editor id)
         * getValue:
         *      Joomla.editors.instances['jform_articletext'].getValue();
         * setValue:
         *      Joomla.editors.instances['jform_articletext'].setValue('Joomla! rocks');
         * replaceSelection:
         *      Joomla.editors.instances['jform_articletext'].replaceSelection('Joomla! rocks')
         *
         * *********************************************************
         * ANY INTERACTION WITH THE EDITORS SHOULD USE THE ABOVE API
         * *********************************************************
         */
    };
```
### Testing Instructions

Apply this patch
Go to back end and edit/create an article
On purpose there are all the editors in that page
Try the buttons in all of them. Cool eh?
### Documentation Changes Required

The comment above comes from core.js but needs to be published also in the form->editor page.
### TL;DR

Some images then:
screen shot 2016-10-26 at 01 13 16

screen shot 2016-10-26 at 01 14 23

Calling @Fedik @andrepereiradasilva @ggppdk @mbabker

---
This change is [Reviewable](https://reviewable.io/reviews/joomla/joomla-cms/12561)

### CODE REVIEW

**Comment by dgt41** [comment]
@okonomiyaki3000 can you check the changes I did in codemirror here?

**Comment by okonomiyaki3000** [comment]
I'm not sure what this `media/system/js/editor-codemirror.js` is about but I don't like it.

**Comment by dgt41** [comment]
it's just the initiator script, this way there is no inline script at all

**Comment by okonomiyaki3000** [comment]
I don't think there's anything wrong with inline script. The problem with this is that it seems to be part of Joomla core instead of the CodeMirror plugin and it can't be overridden like the current version can (via layouts).

**Comment by dgt41** [comment]
@okonomiyaki3000 not quite true, you can still use the layouts although a tiny script will be missing (the one that comes from code mirror.php). But if it breaks B/C I can revert it and redo this in J4

**Comment by okonomiyaki3000** [comment]
Sorry, I can't get behind putting files used by the editors in `media/system/js/`. They just don't belong there. The editors should be thought of as optional plugins, decoupled from the Joomla core.

**Comment by dgt41** [comment]
@okonomiyaki3000 I will revert that part, but this PR is NOT about the initialisation of the editors, it solves a very fundamentally wrong architectural concept that limits joomla to only one editor per page...

**Comment by okonomiyaki3000** [comment]
@dgt41 Yes, I'm aware of the limitation. It limits Joomla to only one type of editor per page. You may still have multiple instances of the same type. It's quite a glaring flaw and I'm glad you're fixing it but we must first do no harm.

**Comment by dgt41** [comment]
@ggppdk can you test this one?

**Comment by ggppdk** [comment]
@dgt41 

will do tomorrow, thanks for the work on this one

**Comment by dgt41** [comment]
@okonomiyaki3000 I've restored the inline initialisation script for codemirror...

**Comment by okonomiyaki3000** [comment]
I don't know, I'm still seeing `media/system/js/editor-codemirror.js`. What about other editors?

**Comment by dgt41** [comment]
> What about other editors?

Other editors also using static scripts in `media/editors/whatever/js`. Also code mirror should move to static script in J4

**Comment by okonomiyaki3000** [comment]
Isn't `media/editors/whatever/js` just the editor's distro code? Is Joomla-specific code meant to go in there too?

**Comment by dgt41** [comment]
Right now the folders ``media/editor/whatever` are maintained/manipulated by joomla, in J4 the aim is that foreign code should live in `media/vendor/` folder and there should be an automated way to manage the updates (code is already written for that).
So here is a glimpse from the future (j4):
screen shot 2016-10-31 at 09 28 58

**Comment by okonomiyaki3000** [comment]
I see, that makes sense I guess. Does that mean that the libraries under `vendor` must be used as distributed? What if we prefer custom builds?

**Comment by dgt41** [comment]
@okonomiyaki3000 I guess this will manage only the joomla distributed code, I have no clue if this will be extended somehow for 3rd PD (my guess is no, 3rd PD could provide their custom code in their own directory). Or you can manipulate or override the grunt script and get whatever you want, with the added step to re-run the script every time joomla updates these assets...

**Comment by infograf768** [comment]
Conflicts

File: administrator/components/com_content/models/forms/article.xml [hunk 1]
```diff
@@ -28,6 +28,24 @@
 			label="COM_CONTENT_FIELD_ARTICLETEXT_LABEL" description="COM_CONTENT_FIELD_ARTICLETEXT_DESC"
 			filter="JComponentHelper::filterText" buttons="true" />
 
+<field
```
Comment (yvesh) [inline administrator/components/com_content/models/forms/article.xml]: @dgt41 Can you look into the formatting here? The indentation seems to be broken, same for closing tag. And you have some white space at the label.

File: administrator/components/com_content/views/articles/tmpl/modal.php [hunk 2]
```diff
@@ -111,8 +123,9 @@
 							<span class="<?php echo $iconStates[$this->escape($item->state)]; ?>"></span>
 						</td>
 						<td>
-							<a href="javascript:void(0);" onclick="if (window.parent) window.parent.<?php echo $this->escape($function); ?>('<?php echo $item->id; ?>', '<?php echo $this->escape(addslashes($item->title)); ?>', '<?php echo $this->escape($item->catid); ?>', null, '<?php echo $this->escape(ContentHelperRoute::getArticleRoute($item->id, $item->catid, $item->language)); ?>', '<?php echo $this->escape($lang); ?>', null);">
-								<?php echo $this->escape($item->title); ?></a>
+							<a class="select-link" href="javascript:void(0)" data-function="<?php echo $this->escape($onclick); ?>" data-id="<?php echo $item->id; ?>" data-title="<?php echo $this->escape(addslashes($item->title)); ?>" data-cat-id="<?php echo $this->escape($item->catid); ?>" data-uri="<?php echo $this->escape(ContentHelperRoute::getArticleRoute($item->id, $item->catid, $item->language)); ?>" data-language="<?php echo $this->escape($lang); ?>">
```
Comment (yvesh) [inline administrator/components/com_content/views/articles/tmpl/modal.php]: Any way to make this more readable and less long? :)

Comment (mbabker) [inline administrator/components/com_content/views/articles/tmpl/modal.php]: Not use as many attributes or start formatting our HTML the same way our XML files are formatted (and I find THAT ugly and painful to keep up with personally).

File: media/com_contact/js/admin-contacts-modal.js [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+ * @copyright  Copyright (C) 2005 - 2016 Open Source Matters, Inc. All rights reserved.
+ * @license    GNU General Public License version 2 or later; see LICENSE.txt
+ */
+(function() {
+	"use strict";
+	/**
+	 * Javascript to insert the link
+	 * View element calls jSelectContact when a contact is clicked
+	 * jSelectContact creates the link tag, sends it to the editor,
+	 * and closes the select frame.
```
Comment (yvesh) [inline media/com_contact/js/admin-contacts-modal.js]: Actually it would be nice to have since tags in js too.. Any thoughts on that?

File: media/com_contact/js/admin-contacts-modal.js [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+		}
+
+		editor = Joomla.getOptions('xtd-contacts').editor;
+
+		if (lang !== '')
+		{
+			hreflang = ' hreflang = "' + lang + '"';
+		}
+
+		tag = '<a' + hreflang + ' href="' + link + '">' + title + '</a>';
```
Comment (yvesh) [inline media/com_contact/js/admin-contacts-modal.js]: Personally i would prefer when you do a document.createElement('a') here etc. then hardcoding a link into a string.

File: media/com_contact/js/admin-contacts-modal.js [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+		}
+
+		window.parent.jModalClose();
+	};
+
+	document.addEventListener('DOMContentLoaded', function(){
+		// Get the elements
+		var elements = document.querySelectorAll('.select-link');
+
+		for(var i = 0, l = elements.length; l>i; i++) {
```
Comment (yvesh) [inline media/com_contact/js/admin-contacts-modal.js]: Just out of curiosity, why are you saving elements length in a variable? ;) cs btw.

File: media/com_contact/js/admin-contacts-modal.js [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+		if (!Joomla.getOptions('xtd-contacts')) {
+			// Something went wrong!
+			window.parent.jModalClose();
+			return false;
+		}
+
+		editor = Joomla.getOptions('xtd-contacts').editor;
+
+		if (lang !== '')
+		{
```
Comment (dgt41) [inline media/com_contact/js/admin-contacts-modal.js]: copy-paste...

File: media/com_contact/js/admin-contacts-modal.js (hunk 4 above)
Comment (dgt41) [inline media/com_contact/js/admin-contacts-modal.js]: Too fast, we do support IE8, so not possible

File: media/com_content/js/admin-article-pagebreak.js [hunk 7]
```diff
@@ -0,0 +1,37 @@
...
+		}
+
+		/** Get the pagebreak toc alias -- not inserting for now **/
+		/** don't know which attribute to use... **/
+		alt = document.getElementById('alt').value;
+
+		title  = (title != '') ? 'title="' + title + '"' : '';
+		alt    = (alt != '') ? 'alt="' + alt + '"' : '';
+
+		tag = '<hr class="system-pagebreak" ' + title + ' ' + alt + '/>';
```
Comment (dgt41) [inline media/com_content/js/admin-article-pagebreak.js]: IE8

File: media/com_content/js/admin-article-readmore.js [hunk 8]
```diff
@@ -0,0 +1,31 @@
...
+	}
+
+	var options, content;
+
+	options = window.parent.Joomla.getOptions('xtd-readmore');
+
+	content = (new Function('return ' + options.editor))();
+
+	if (content.match(/<hr\s+id=("|')system-readmore("|')\s*\/*>/i)) {
+		alert(options.exists);
```
Comment (dgt41) [inline media/com_content/js/admin-article-readmore.js]: it's what was there, what else can it be?

**Comment by yvesh** [comment]
@dgt41 Would be nice to have some javascript tests too. :+1: Ping @Ruchiranga

File: media/com_contact/js/admin-contacts-modal.js (hunk 5 above)
Comment (Fedik) [inline media/com_contact/js/admin-contacts-modal.js]: need for speed 😄

File: media/com_contact/js/admin-contacts-modal.js (hunk 4 above)
Comment (yvesh) [inline media/com_contact/js/admin-contacts-modal.js]: Hm document.createElement should work in IE8?

File: administrator/components/com_content/models/forms/article.xml (hunk 1 above)
Comment (dgt41) [inline administrator/components/com_content/models/forms/article.xml]: @yvesh this whole section is here only for testing, will be and should be removed before merging, thus the wrong indentation

**Comment by dgt41** [comment]
@yvesh you are right, now that all these are not inline some tests could be written, any help is welcome 😄

File: media/com_content/js/admin-article-readmore.js (hunk 8 above)
Comment (yvesh) [inline media/com_content/js/admin-article-readmore.js]: I don't know why it is there any ways, but not a fan of alerts at all ;) Maybe use [Joomla.renderMessages](https://github.com/joomla/joomla-cms/blob/staging/media/system/js/core-uncompressed.js#L250) instead?

File: media/com_content/js/admin-article-pagebreak.js (hunk 7 above)
Comment (yvesh) [inline media/com_content/js/admin-article-pagebreak.js]: Hm it should work fine in IE8? Shouldn't it? We use document.createElement in a lot of other JS files

File: media/com_content/js/admin-article-readmore.js (hunk 8 above)
Comment (dgt41) [inline media/com_content/js/admin-article-readmore.js]: I guess is the same story as the buttons of the toolbar, where also an alert is utilised...
Honestly I don't mind changing it to Joomla.renderMessages()

File: media/com_content/js/admin-article-pagebreak.js (hunk 7 above)
Comment (dgt41) [inline media/com_content/js/admin-article-pagebreak.js]: Actually it shouldn't be any problem, just checked...

**Comment by Ruchiranga** [comment]
Bit busy this month. I'll see what I can do when I get some free time :wink:

File: administrator/components/com_content/views/articles/tmpl/modal.php (hunk 2 above)
Comment (dgt41) [inline administrator/components/com_content/views/articles/tmpl/modal.php]: @mbabker would you prefer something like:
```php
							<?php $attribs = 'data-function="' . $this->escape($onclick) . '"'
 								. ' data-id="' . $item->id . '"'
 								. ' data-title="' . $this->escape(addslashes($item->title)) . '"'
 								. ' data-cat-id="' . $this->escape($item->catid) . '"'
 								. ' data-uri="' . $this->escape(ContentHelperRoute::getArticleRoute($item->id, $item->catid, $item->language)) . '"'
 								. ' data-language="' . $this->escape($lang) . '"';
 							?>
 							<a class="select-link" href="javascript:void(0)" <?php echo $attribs; ?>>
```

Comment (mbabker) [inline administrator/components/com_content/views/articles/tmpl/modal.php]: Actually, for this you could just as usually use `JHtml::_('link', 'javascript:void(0)', $this->escape($item->title), $attribsArray);` and it not be unbearable.  What you have works too, I'm just honestly not the biggest fan of some of our code style rules with line concatenation and splitting elements on separate lines in funny places.

Comment (mbabker) [inline administrator/components/com_content/views/articles/tmpl/modal.php]: Until recently, I seriously thought our `iframe` and `link` HTML helpers were silly as all can be.  But then I started thinking that they're actually pretty useful for elements where you're adding a lot of data in some form (in this case data attributes) so you can throw those elements in an array and pass them into a function which converts it to the proper string.  End result is still the same; clean PHP code and clean HTML output.

**Comment by ggppdk** [comment]
Is this ready for testing ? 

In Joomla article form , if i use tinymce editor, i am getting (firefox console):

> SyntaxError: expected expression, got '<'[Learn more]index.php:1

**Comment by dgt41** [comment]
Sorry @ggppdk I've moved some files around and forgot to fix the paths, now should be fine

**Comment by ggppdk** [comment]
@dgt41 

please re-open
after your latest changes, i had re-tested but i was still getting

> SyntaxError: expected expression, got '<'[Learn more]index.php:1

Some attempt to get JS code actually returns html code, i was going to look more to it, and find for which URL it is happening

**Comment by ggppdk** [comment]
Thanks,

i will post back, with the URL, that is causing this

I had found the URL, but i had not find the PHP file that was adding it

**Comment by dgt41** [comment]
@ggppdk let me know what's the URL so I could test that

File: plugins/editors/tinymce/tinymce.php [hunk 9]
```diff
@@ -736,7 +751,8 @@ public function onDisplay($name, $content, $width, $height, $col, $row, $buttons
 
 		if (!empty($btnsNames))
 		{
-			JHtml::_('script', 'system/tiny-close.min.js', array('version' => 'auto', 'relative' => true), array('defer' => 'defer'));
+			$modalFix = JHtml::_('script', 'system/tiny-close.min.js', false, true, true, false, true);
```
Comment (ggppdk) [inline plugins/editors/tinymce/tinymce.php]: This file 'system/tiny-close.min.js' is not found by
JHtml::_('script', 'system/tiny-close.min.js', false, true, **$pathOnly=true**
thus it returns null into $modalFix

and  then you get:
JFactory::getDocument()->addScript(null, "text/javascript", true, false);

which adds:
```
<script src="" defer></script>
```

thus current page which is HTML is retrieved as a JS file and then browser tries to parse it and you get error:
SyntaxError: expected expression, got '<'1index.php:1

Comment (mbabker) [inline plugins/editors/tinymce/tinymce.php]: 1) The file moved to `media/editors/tinymce/js`

2) This should be "fixed" to use the 3.7 API signature again

With both of those things should be fine.

Comment (dgt41) [inline plugins/editors/tinymce/tinymce.php]: @ggppdk @mbabker fixed it.

**Comment by ggppdk** [comment]
Tested, looks good, but can you rebase this PR ?

there have been changes to e.g. addStyleSheet, etc functions

**Comment by dgt41** [comment]
@ggppdk I just pressed that shiny button `update branch`. Hopefully everything still ok 😄

File: media/com_contact/js/admin-contacts-modal.js (hunk 3 above)
Comment (dgt41) [inline media/com_contact/js/admin-contacts-modal.js]: @yvesh so what is the `@since` supposed to be for all these scripts, because ok now they are static files but the code comes from way back?

File: media/com_content/js/admin-article-pagebreak.js (hunk 7 above)
Comment (dgt41) [inline media/com_content/js/admin-article-pagebreak.js]: @yvesh checking this again, I am not in favour as we are passing a string to the editor and I have no clue if all editors out there will be happy with an element instead of a string

File: media/com_content/js/admin-article-readmore.js [hunk 10]
```diff
@@ -0,0 +1,31 @@
+/**
+ * @copyright  Copyright (C) 2005 - 2016 Open Source Matters, Inc. All rights reserved.
+ * @license    GNU General Public License version 2 or later; see LICENSE.txt
+ */
+
+"use strict";
```
Comment (Fedik) [inline media/com_content/js/admin-article-readmore.js]: please do not use "globaly" in the file, 
wrap it in to an anonymous function or place inside `window.insertReadmore = function(editor) {`

because if someday, someone merge the script files on the page in to single one, then such thing could "break" code in the merged file

Comment (dgt41) [inline media/com_content/js/admin-article-readmore.js]: @Fedik good point! will fix them

File: media/editors/tinymce/js/tinymce.js [hunk 11]
```diff
@@ -60,22 +62,41 @@
...
+
+			/** Register the editor's instance to Joomla Object */
+			Joomla.editors.instances[element.id] = {
+				// Required by Joomla's API for the XTD-Buttons
+				'getValue': function () { return tinyMCE.editors[this.id].getContent(); },
+				'setValue': function (text) { return tinyMCE.editors[this.id].setContent(text); },
+				'replaceSelection': function (text) { return tinyMCE.editors[this.id].execCommand('mceInsertContent', false, text); },
+				// Some extra instance dependent
+				'id': element.id,
+				'instance': ed,
```
Comment (Fedik) [inline media/editors/tinymce/js/tinymce.js]: very good idea 😉

File: media/editors/tinymce/js/tinymce.js [hunk 12]
```diff
@@ -60,22 +62,41 @@
...
 
-			tinyMCE.init(options);
+			// Create a new instance
+			var ed = new tinyMCE.Editor(element.id, options, tinymce.EditorManager);
+			ed.render();
+
+			/** Register the editor's instance to Joomla Object */
+			Joomla.editors.instances[element.id] = {
+				// Required by Joomla's API for the XTD-Buttons
+				'getValue': function () { return tinyMCE.editors[this.id].getContent(); },
```
Comment (Fedik) [inline media/editors/tinymce/js/tinymce.js]: since you already have an `instance` property, you better use `return this.instance.getContent();`

Comment (dgt41) [inline media/editors/tinymce/js/tinymce.js]: Done!

**Comment by jeckodevelopment** [comment]
@dgt41 can you please fix conflicts here?

`media/system/js/tinymce-init.js`
`media/system/js/tinymce-init.min.js`

**Comment by dgt41** [comment]
@jeckodevelopment done

**Comment by dgt41** [comment]
Any interest to get this merged or should I close it?

**Comment by ggppdk** [comment]
I ll test after you re-fix conflicts, and re-test after that if needed )

**Comment by ggppdk** [comment]
It works, but 
- new configuration that allows tinyMCE configurations per per access level added by recent PR is not used
- the editor always start in OFF state i have to click "toggle editor" to show it

I think we need to wait for this PR to be tested:
https://github.com/joomla/joomla-cms/pull/13387

and then update / test this one ?

**Comment by anibalsanchez** [comment]
I have tested this item :red_circle: unsuccessfully on 40ba81f64d4b81e596c0f8c278359e405f59b095<br /><br />In the testing steps, there is no tmpl/default.php... so I guess it has to be tmpl/edit.php

Buttons only work in the last editor.
<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/12561">issues.joomla.org/tracker/joomla-cms/12561</a>.</sub>

**Comment by dgt41** [comment]
@ggppdk yes that needs to be pulled as well
@anibalsanchez I just tested it again and works fine here, I also updated the description and the steps required to test this..


Anyways let's wait for the https://github.com/joomla/joomla-cms/pull/13387 to be merged and then I will re update this one

**Comment by ggppdk** [comment]
@dgt41 
PR #13387 that also deal with tinyMCE was merged,
this PR can be updated

**Comment by rvbgnu** [comment]
Great one to test, once updated ;-)

**Comment by dgt41** [comment]
@ggppdk @rvbgnu @anibalsanchez re synced the repo, can you give it a go?

**Comment by rvbgnu** [comment]
I have tested this item :white_check_mark: successfully on 125eedaec4409a1203f7be0592f512f580e50f41<br /><br />Thanks Dimitri!

Just a precision, as I am not sure to be able to reproduce the issue, compare to your images: I don't have the any content in the two additional editors when I open an existing article. As I understand a bit the two code changes to get them, I supposed it is normal. I may suggest to edit your test instruction to remove the confusion.

Otherwise, I have the buttons working properly in the other two editors. So I marked this test successful (Please remove it if I'm wrong).

<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/12561">issues.joomla.org/tracker/joomla-cms/12561</a>.</sub>

**Comment by dgt41** [comment]
> As I understand a bit the two code changes to get them, I supposed it is normal

Yes, you are right, the extra editors are dummies, so any data you've put into them will be lost after save (there is no code to manipulate it and save it in the database).
Thanks you for testing this, really appreciated! 👍

**Comment by ggppdk** [comment]
@dgt41 

Thanks for your works, i will make a detailed test at end of week, now little time

**Comment by anibalsanchez** [comment]
I have tested this item :white_check_mark: successfully on 35502139a8e9eba2096779ce050fc92f2dc4c90b<br /><br />Test OK<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/12561">issues.joomla.org/tracker/joomla-cms/12561</a>.</sub>

**Comment by dgt41** [comment]
@ggppdk @rvbgnu @infograf768 can we have one more test here, so we can RTC this?

**Comment by rvbgnu** [comment]
So nobody for this test ?

Or should I re-test it (changes since 08/01/2017)??<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/12561">issues.joomla.org/tracker/joomla-cms/12561</a>.</sub>

**Comment by dgt41** [comment]
@rvbgnu if you can re-test this, will be great

**Comment by rvbgnu** [comment]
So I've tried, but it doesn't change before or after the patch. The TinyMCe also had an update. I have to reset with repository copy and re-install I think...<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/12561">issues.joomla.org/tracker/joomla-cms/12561</a>.</sub>

**Comment by dgt41** [comment]
> but it doesn't change before or after the patch

@rvbgnu the article form *doesn't* have the extra editors anymore. That needs to be done by editing the xml and the view files. The initial approach had those extra editors as a way to make testing easier...

**Comment by laoneo** [comment]
I have tested this item :white_check_mark: successfully on 35502139a8e9eba2096779ce050fc92f2dc4c90b<br /><br />If this is in, then we can offer the different editors attributes on com_fields.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/12561">issues.joomla.org/tracker/joomla-cms/12561</a>.</sub>

**Comment by dgt41** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/12561">issues.joomla.org/tracker/joomla-cms/12561</a>.</sub>

**Comment by dgt41** [comment]
Note to whomever might merge this: please ping to update the documentation for the editor input field, etc

**Comment by zero-24** [comment]
> Note to whomever might merge this: please ping to update the documentation for the editor input field, etc

@dgt41 ping :P

**Comment by svenbluege** [comment]
I just grabbed the latest build and run into an issue with having multiple editors on the same page. Seems like only one of them gets initialized. I have three of them in my example. The `Joomla.editors.instances` object contains just the first editor instance. The `tinyMCE.editors` shows one as well. If I click the "toggle editor" button, this field shows up in `tinyMCE.editors`. 

image

This can also be reproduced by changing some form elements of the com_content/forms/article.xml to type editor: 

image

**Comment by dgt41** [comment]
> I just grabbed the latest build and run into an issue with having multiple editors on the same page. 

That's impossible (each tinyMCE or code mirror or none has it's own instance) according to the code: https://github.com/joomla/joomla-cms/blob/staging/media/editors/tinymce/js/tinymce.js#L65-L79
Can you type Joomla.editors and press enter in your browsers console? Do you have all the instances there?

**Comment by dgt41** [comment]
@svenbluege actually there is a bug, but here is the fix: https://github.com/joomla/joomla-cms/pull/14194

Thanks for testing!

**Comment by svenbluege** [comment]
@dgt41 works for me. Thanks!

**Comment by infograf768** [comment]
It looks like this has broken inserting menu item urls
https://github.com/joomla/joomla-cms/issues/14890

Please look @dgt41

**Comment by dgt41** [comment]
@infograf768 check #14902
